projects/plugins/boost/app/modules/optimizations/page-cache/data-sync/Page_Cache_Entry.php

```diff
@@ -27,19 +27,72 @@ public function set( $value ) {
 	}
 
 	/**
-	 * Sanitizes the given value, ensuring that it is a comma-separated list of unique, trimmed strings.
+	 * Sanitizes the given value, ensuring that it is list of valid patterns.
 	 *
 	 * @param mixed $value The value to sanitize.
 	 *
-	 * @return string The sanitized value, as a comma-separated list of unique, trimmed strings.
+	 * @return string The sanitized value, as a list.
 	 */
 	private function sanitize_value( $value ) {
 		if ( is_array( $value ) ) {
-			$value = array_values( array_unique( array_filter( array_map( 'trim', $value ) ) ) );
+			$value = array_values( array_unique( array_filter( array_map( 'trim', array_map( 'strtolower', $value ) ) ) ) );
+
+			$home_url = home_url( '/' );
+
+			foreach ( $value as &$path ) {
+				// Strip home URL (both secure and non-secure).
+				$path = str_ireplace(
+					array(
+						$home_url,
+						str_replace( 'http:', 'https:', $home_url ),
+					),
+					array(
+						'/',
+						'/',
+					),
+					$path
+				);
+
+				// Remove double shashes.
+				$path = str_replace( '//', '/', $path );
+
+				// Make sure there's a leading slash.
+				$path = '/' . ltrim( $path, '/' );
+
+				// Fix up any wildcards.
+				$path = $this->sanitize_wildcards( $path );
+			}
 		} else {
 			$value = array();
 		}
 
 		return $value;
 	}
+
+	/**
+	 * Sanitize wildcards in a given path.
+	 *
+	 * @param string $path The path to sanitize.
+	 * @return string The sanitized path.
+	 */
+	private function sanitize_wildcards( $path ) {
+		if ( ! $path ) {
+			return '';
+		}
+
+		$path_components = explode( '/', $path );
+		$arr             = array(
+			'.*'   => '(.*)',
+			'*'    => '(.*)',
+			'(*)'  => '(.*)',
```

Comment:
`Cheese*` is a valid regexp which will match chees, cheese, cheeseeee, etc.

However, it seems like a pattern that users are very unlikely to want.

I think I'm ok with the guard-rails you've put here, because for most users this will put them on the right path. But I did want to note this will make some esoteric regexps impossible to enter.